front/src/app/todo-list/todo-list.component.html

```diff
@@ -1,32 +1,76 @@
 <div [formGroup]="radioBtnGroup" class="task-list">
   <mat-radio-group formControlName="activeTaskList" class="task-list__group">
     <a
-      *ngFor="let todoList of allTaskLists?.lists" 
+      *ngFor="let todoList of allTaskLists?.lists; let taskListIndex = index" 
       [routerLink]="['/todo']"
       [queryParams]="{list: todoList.id}"
       class="task-list__link"
     >
-      <mat-radio-button [value]="todoList.id" class="task-list__list">{{todoList.title}}</mat-radio-button>
+      <mat-radio-button [value]="todoList.id" class="task-list__list">
+        <div class="task-list__task-group">
+          <span [ngClass]="{'hide-element': activeTaskGroupId == taskListIndex}">
+            {{todoList.title}}
+          </span>
+          <input
+            [ngClass]="{'hide-element': activeTaskGroupId != taskListIndex}"
+            #renameTaskGroupFormField
+            matInput
+            name="input{{taskListIndex}}"  
+            type="text"
+            (keydown.enter)="renameTaskGroup(todoList.id, renameTaskGroupFormField.value)"
+            (blur)="activeTaskGroupId = undefined"
+          >
+        </div>
+        <div>
+          <button 
+            [ngClass]="{'hide-element': activeTaskGroupId == taskListIndex}" 
+            [matMenuTriggerRestoreFocus]="false" 
+            [matMenuTriggerFor]="menu" 
+            mat-icon-button
+            disableRipple="true"
+            class="task-list__showing-control-menu"
+          >
+            <mat-icon>more_horiz</mat-icon>

```

Comment:
If I'm not mistaken, when the user click on this button the Task List is getting opened at that moment (so that tasks of this list appears on the screen).

That's not a proper behavior, see `event.stopPropagation` to prevent that